packages/react/src/components/Chat/ChatItem.tsx

```diff
@@ -89,23 +90,62 @@ class ChatItem extends UIComponent<ReactProps<ChatItemProps>, any> {
   }
 
   private renderChatItem(styles: ComponentSlotStylesPrepared) {
-    const { message, gutter, contentPosition } = this.props
+    const { gutter, contentPosition } = this.props
     const gutterElement =
       gutter &&
       Box.create(gutter, {
         defaultProps: { className: ChatItem.slotClassNames.gutter, styles: styles.gutter },
       })
 
+    const messageElement = this.getMessageAugmentedWithChatMessageProps(styles)
+
     return (
       <>
         {contentPosition === 'start' && gutterElement}
-        {Box.create(message, {
-          defaultProps: { className: ChatItem.slotClassNames.message, styles: styles.message },
-        })}
+        {messageElement}
         {contentPosition === 'end' && gutterElement}
       </>
     )
   }
+
+  getMessageAugmentedWithChatMessageProps = styles => {
+    const { message, attached } = this.props
+    const messageElement = Box.create(message, {
+      defaultProps: {
+        className: ChatItem.slotClassNames.message,
+        styles: styles.message,
+      },
+    })
+
+    // the box element is ChatMessage
+    if (this.checkIfElementIsChatMessage(messageElement)) {
+      return this.cloneElementWithCustomProps(messageElement, { attached })
+    }
+
+    // the children element is ChatMessage
+    if (this.checkIfElementIsChatMessage(messageElement, 'children')) {
+      return this.cloneElementWithCustomProps(messageElement, { attached }, 'children')
+    }
+
+    // the content element is ChatMessage
+    if (this.checkIfElementIsChatMessage(messageElement, 'content')) {
+      return this.cloneElementWithCustomProps(messageElement, { attached }, 'content')
+    }
+    return messageElement
+  }
+
+  checkIfElementIsChatMessage = (element, prop?) => {
+    return _.get(element, `${prop ? `props.${prop}.` : ''}type.__isChatMessage`)
```

Comment:
I would suggest to make this method's logic as simple as possible - then it would be much easier to reason of what this method is supposed to do at the place of usage:

```jsx
// just single parameter
isChatMessageElement = (element) => ...
```

As a next step, we might notice that this method relies on the aspects that are provided by another type, i.e. `ChatMessage`'s `__isChatMessage` static prop, which name might change. Thus, I would suggest to put this check as part of the `ChatMessage` component's logic:

```jsx
class ChatMessage extends ... {
  static isChatMessageElement(element) => ...
}
```

And, finally - while I do see the benefit of introducing second arg to original method to avoid duplication of prop fetching logic, we might just want to introduce a dedicated method for props fetching for that:

```jsx
getElementProp = (element, propName) => { .. }
```

once again, with preserving the benefit of reducing duplication, we would have yet additional one: this method is quite easy to reason about on the client's side.

As a result, we might end with something like:

```jsx
 if (ChatMessage.isTypeOfElement(messageElement) { .. }
 if (ChatMessage.isTypeOfElement(getElementProp(messageElement, 'content')) { .. }
```